gcloud/search/document.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Document."""
+
+import datetime
+
+import six
+
+from gcloud._helpers import UTC, _RFC3339_MICROS
+from gcloud.exceptions import NotFound
+
+
+class StringValue(object):
+    """Values hold individual values for a given field"""
+
+    value_type = 'string'
+
+    def __init__(self, string_value, string_format=None, language=None):
+        self.string_value = string_value
+        self.string_format = string_format
+        self.language = language
+
+
+class NumberValue(object):
+    """Values hold individual values for a given field"""
+
+    value_type = 'number'
+
+    def __init__(self, number_value):
+        self.number_value = number_value
+
+
+class TimestampValue(object):
+    """Values hold individual values for a given field"""
+
+    value_type = 'timestamp'
+
+    def __init__(self, timestamp_value):
+        self.timestamp_value = timestamp_value
+
+
+class GeoValue(object):
+    """Values hold individual values for a given field"""
+
+    value_type = 'geo'
+
+    def __init__(self, geo_value):
+        self.geo_value = geo_value
+
+
+class Field(object):
+    """Fields hold values for a given document"""
+
+    def __init__(self, name):
+        self.name = name
+        self.values = []
+
+    def add_value(self, value, **kw):
+        """Add a value to the field.
+
+        Selects type of value instance based on type of ``value``.
+
+        :type value: string, integer, float, datetime, or tuple (float, float)
+        :param value: the field value to add.
+
+        :param kw: extra keyword arguments to be passed to the value instance
+                   constructor.
+
+        :raises: ValueError if unable to match the type of ``value``.
+        """
+        if isinstance(value, six.string_types):
+            self.values.append(StringValue(value, **kw))
```

Comment:
We pass `**kw`` to all of these but `StringValue` is the only one that can use it, right?